src/count-cats.js

```diff
@@ -1,6 +1,21 @@
 const CustomError = require("../extensions/custom-error");
 
-module.exports = function countCats(/* matrix */) {
-  throw new CustomError('Not implemented');
-  // remove line with error and write your code here
+module.exports = function countCats(cats) {
+  let total = 0;
+  if (!Array.isArray(cats) || cats.length == 0) {
+    return total;
+  }
+
+  for (const element of cats) {
+    if (!Array.isArray(element) || element.length == 0) {
+      break;
+    }
+    total += element.reduce(function(sum, current) {
```

Comment:
I don't think this solution is an optimal one. Apparently, there is a better way on how to do it.